res/controllers/Gemini-G4V-scripts.js

```diff
@@ -0,0 +1,562 @@
+            inKey: "loop_double",
+            outKey: "eject",
+        }),
+        pad7: new components.Button({
+            midi: [0x8F + midiChannel, 0x0F],
+            group: this.currentDeck,
+            inKey: "loop_move_1_backward",
+            outKey: "eject",
+        }),
+        pad8: new components.Button({
+            midi: [0x8F + midiChannel, 0x10],
+            group: this.currentDeck,
+            inKey: "loop_move_1_forward",
+            outKey: "eject",
+        }),
+    });
+
+    // Pad selection buttons (not connected a CO)
+    this.padModeBtn = new components.Component({
+        deck: this,
+        input: function(_channel, control, value, _status, _group) {
+            if (!value) { return; }
+            switch (control) {
+            case 0x1B:
+                this.deck.padMode(g4v.Deck.prototype.modes.cue);
+                break;
+            case 0x1C:
+                this.deck.padMode(g4v.Deck.prototype.modes.autoLoop);
+                break;
+            case 0x1D:
+                this.deck.padMode(g4v.Deck.prototype.modes.sample);
+                break;
+            case 0x1E:
+                this.deck.padMode(g4v.Deck.prototype.modes.loopRoll);
+                break;
+            case 0x1F:
+                // Not mapped, for future use
+                // TODO - Four buttons to enable effect racks
+                break;
+            case 0x20:
+                // Not mapped, for future use
+                break;
+            case 0x21:
+                this.deck.padMode(g4v.Deck.prototype.modes.beatJump);
+                break;
+            case 0x22:
+                this.deck.padMode(g4v.Deck.prototype.modes.manualLoop);
+                break;
+            }
+        },
+    });
+    // Deck togle button (not connected to CO)
+    this.deckToggleBtn = new components.Component({
+        deck: this,
+        input: function(_channel, _control, value, _status, _group) {
+            if (!value) { return; }
+            this.deck.toggle();
+            const index = this.deck.deckNumbers.indexOf(parseInt(
+                script.channelRegEx.exec(this.deck.currentDeck)[1]));
+            if (index === 1) { midi.sendShortMsg(0x8F + midiChannel, 0x26, 0x7F); } else { midi.sendShortMsg(0x8F + midiChannel, 0x26, 0x00); }
+        },
+    });
+
+    this.jogTopBtn = new components.Component({
+        input: function(_channel, _control, value, _status, _group) {
+            switch (value) {
+            case 0x7f: {
+                const intervalsPerRev = 400;
+                const rpm = 30+1/3;
+                const alpha = (1.0/8);
+                const beta = (alpha / 32);
+                engine.scratchEnable(script.deckFromGroup(this.group), intervalsPerRev, rpm, alpha, beta);
+                break;
+            }
+            case 0x00:
+                engine.scratchDisable(script.deckFromGroup(this.group), 1);
+                break;
+            }
+        },
+    });
```

Comment:
use `components.Button` instead and then use ` if (this.isPress(...))` instead of switching on the value.